morphir/sdk/core/src/morphir/sdk/Dict.scala

```diff
@@ -60,14 +93,58 @@ object Dict {
       case xs  => DictImpl(xs.toMap)
     }
 
-  def toList[K, V](dict: Dict[K, V]): List[(K, V)] =
+  /* Transform */
+  def map[K, V, B](f: K => V => B)(dict: Dict[K, V]): Dict[K, B] = dict match {
+    case EmptyDict         => empty
+    case DictImpl(wrapped) => DictImpl(wrapped.map(x => (x._1, f(x._1)(x._2))))
+  }
+
+  def foldl[K, V, B](initValue: B)(f: B => (K, V) => B)(dict: Dict[K, V]): B = dict match {
```

Comment:
This type signature doesn't look right. The Elm version looks like this:
```elm
foldl : (k -> v -> b -> b) -> b -> Dict k v -> b
```
So the direct translation is:
```scala
def foldl[K, V, B](f: K => V => B => B)(initValue: B)(dict: Dict[K, V]): B
```